text/0000-precise-pre-release-dependencies.md

```diff
@@ -0,0 +1,127 @@
+| `a.b.c`         | `a.b.c`            | `x.y.z`        | ✅                        | ✅                                  |
+| `a.b.c`         | `a.b.c`            | `x.y.z-pre0`   | ❌                        | ✅                                  |
+| `a.b.c`         | `x.y.z-pre0`       | `x.y.z-pre1`   | ❌                        | ✅                                  |
+| `a.b.c-pre0`    | `a.b.c-pre0`       | `a.b.c-pre1`   | ✅¹                       | ✅                                  |
+| `a.b.c`         | `a.b.c`            | `a.b.c-pre0`   | ❌                        | ❌                                  |
+| `a.b.c-pre0`    | `a.b.c-pre0`       | `x.y.z`        | ❌²                       | ✅                                  |
+
+✅: Will upgrade
+
+❌: Will not upgrade
+
+¹For backwards compatibility with Cargo's current behaviour
+
+²Emits a warning
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Pre-release versions are not easily auditable when they are only specified in the lock file.
+  A change that makes use of a pre-release version may not be noticed during code review as reviewers don't always check for changes in the lock file.
+- Library crates that require a pre-release version are not well supported (see [future-possibilities])
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The main alternative to this would be to accept that pre-release versions are not very usable and discourage their use.
+Cargo overrides can be used instead using `[patch]`.
+These provide a similar experience to pre-releases, however, they require that the library's code is somehow vendored outside of the registry, usually with git.
+This can cause issues particularly in CI where jobs may have permission to fetch from a private registry but not from private git repositories.
+Resolving issues around not being able to fetch pre-releases from the registry usually wastes a significant amount of time.
+
+Another alternative would be to resolve pre-release versions in `Cargo.toml`s even when a dependency specifies a stable version.
+This is explored in [future-possibilities].
+This would require significant changes to the resolver since the latest compatible version would depend on the versions required by other parts of the dependency tree.
+For that reason, I consider detailing such a change outside of the scope of this RFC.
+
+# Prior art
+[prior-art]: #prior-art
+
+[RFC: Precise Pre-release Deps](https://github.com/rust-lang/rfcs/pull/3263) aims to solve a similar but subtly different issue where `cargo update` opts to upgrade 
+pre-release versions to new pre-releases when one is released.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+It would be nice if dependencies could specify their requirements for pre-release versions.
```

Comment:
Could be you be more specific? I don't understand what kind of requirements you are talking about here.

Reply:
@mercxry 

I've tried to improve this here https://github.com/eopb/rfcs/pull/1/commits/231cf786f5b0e815d1aa41c79da45a0d411df835

Let me know if that helps?

Reply:
Yep sounds good, left some other corrections 👍